Read contributors and projects

MartinAnderson2 opened the pull request:
Wrote get methods on the facade and read requests on the server.

I am new to everything involved in this project, so please let me know if I did something wrong.

Commit:
Added getContributor and getProject methods

Commit:
Added readContributor and readProject requests

GaryHilares commented on src/entrypoints/server.ts:
```diff
@@ -51,6 +51,38 @@ app.post('/createProject', (req, res) => {
   }
 });
 
+app.get('/readContributor', (req, res) => {
```
Good practices for RESTful APIs indicate that routes should use nouns to refer to resources ("readContributor" is not a noun, see https://restfulapi.net/resource-naming/ for more information).

GaryHilares commented on src/entrypoints/server.ts:
```diff
@@ -51,6 +51,38 @@ app.post('/createProject', (req, res) => {
+  }
+});
+
+app.get('/readProject', (req, res) => {
```
Read comment above ("readProject" is not a noun, see https://restfulapi.net/resource-naming/ for more information).

GaryHilares reviewed (changes requested):
Hello Martin! It looks mostly good to me. Can you rename the API endpoints though? Good practices for RESTful indicate that routes should use nouns to refer to resources ("readContributor" is not a noun, see https://restfulapi.net/resource-naming/ for more information).

GaryHilares commented:
I forgot to mention. Could you also try passing the ID using slugs instead of the body of the request?

That is, instead of querying the URL `/contributors/` with the ID as part of the body of the request, try querying the URL `/contributors/:id` (where `:id` is the ID being queried) with nothing special in the body of the request.

I think you should probably be able to do this checking Express's docs, but please do let me know if you are struggling/need additional help.

MartinAnderson2 commented:
I have made the requested changes. Please let me know if you notice anything else.

MartinAnderson2 committed:
Renamed the get requests to simply contributor and project to follow RESTful best naming practices.

Made these requests pass the ID using slugs instead of the body of the request.

MartinAnderson2 commented:
I have made the requested changes. Please let me know if you notice anything else.

GaryHilares reviewed (changes requested):
Can you please add tests for the code that you added?

GaryHilares commented:
Actually, I am going to merge this to enable implementation of the database code. We'll put writing the tests for this in a different issue.
